WikiWatershed/model-my-watershed
# AWS 2-7: UI Updates for Global Analyze Support

## Overview

Implements UI updates to support Global Analyze:

- Disables CONUS-only visual layers when map viewport is outside CONUS
- Disables Select Boundary draw tool when map viewport is outside CONUS, since all boundaries are CONUS only
- Disables NHD and DRB based Rapid Watershed Delineation when outside CONUS
- Adds explicit CONUS check to NHD and DRB based Rapid Watershed Delineation, both in the front- and back-ends
- Removes flat CONUS check on all AoIs, since now we will have global AoIs
- Adds explicit CONUS check to all CONUS-only analyses, both in the front- and back-ends
- Hides non-CONUS analyses when AoI is outside CONUS
- Disables Monitor and Model tabs when AoI is outside CONUS
- Standardizes perimeter definition and usage

Closes #3583 
Closes #3648 
Closes #3649
Closes #3650 
Closes #3651 

### Demo

Layer selector items disabling when out of CONUS, enabling when inside:

[Screencast from 2024-11-02 15-35-01.webm](https://github.com/user-attachments/assets/ae4f0a50-002f-4bbb-8569-5bb26f33fd12)

Draw tools disabling when out of CONUS, enabling when inside:

[Screencast from 2024-11-02 15-36-46.webm](https://github.com/user-attachments/assets/d11f9d85-0d4c-4f58-8780-a95f2bc9960c)

CONUS Analyze and Monitor and Model tabs working as before:

[Screencast from 2024-11-02 15-57-18.webm](https://github.com/user-attachments/assets/8f3d139c-f7b4-4035-9044-e66631e555bc)

Limited Analyze, disabled Monitor and Model tabs when outside CONUS:

[Screencast from 2024-11-02 16-01-39.webm](https://github.com/user-attachments/assets/6853960e-6ac5-49b4-80f1-d92337d71a98)

## Testing Instructions

- Check out this branch and reprovision app and tiler:
    ```
    vagrant reload app tiler --provision
    ```
- Go to http://localhost:8000/
  - [x] Ensure the Select Boundary button is enabled
- Move the map so that the continental United States is no longer visible
  - [x] Ensure the Select Boundary button is disabled, with a message
  - [x] Ensure CONUS-specific visual layers (e.g. NLCD, Soils, etc) are disabled when the map does not show continental United States
- Move the map to scroll the continental United States back into view
  - [x] Ensure the Select Boundary button is enabled again
  - [x] Ensure CONUS-specific visual layers turn back on
- Select a shape within CONUS
  - [x] Ensure Analyze works as before
  - [x] Ensure Monitor and Model tabs have content as before
  - [x] Ensure that Site Storm Model works as before
- Select a shape outside CONUS
  - [ ] Ensure Analyze now only has global streams and global land cover options
    > [!NOTE]
    > Global Streams will not work unless they are installed. This is a very large dataset, and can be ignore for this PR.
  - [x] Ensure that Monitor and Model tabs are disabled
